arch/x86/core/fatal.c

```diff
@@ -109,6 +109,27 @@ FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
 	_SysFatalErrorHandler(reason, pEsf);
 }
 
+
+FUNC_NORETURN void _arch_syscall_oops(void *ssf_ptr)
+{
+	NANO_ESF oops_esf;
+	struct _x86_syscall_stack_frame *ssf;
+
+	ssf = (struct _x86_syscall_stack_frame *)ssf_ptr;
+	memset(&oops_esf, 0, sizeof(NANO_ESF));
```

Comment:
You can just use designated initializers here:

```C
struct _x86_syscall_stack_frame *ssf = ssf_ptr;
NANO_ESF oops_esf = {
    .eip = ssf->eip,
    .cs = ssf->cs,
    .eflags = ssf->eflags,
    .esp = ssf->cs == USER_CODE_SEG ? ssf->esp : 0,
};

_NanoFatalErrorHandler(_NANO_ERR_KERNEL_OOPS, &oops_esf);
```

Other aggregates will be initialized to 0, so you ditch the memset() call (which might even be removed by the compiler, but the code is cleaner anyway).